catalog/app/containers/Bucket/PackageDialog/FilesInput.tsx

```diff
@@ -19,9 +21,29 @@ import useMemoEq from 'utils/useMemoEq'
 import * as Types from 'utils/types'
 
 import EditFileMeta from './EditFileMeta'
+import EditFileName from './EditFileName'
 import * as PD from './PackageDialog'
 import * as S3FilePicker from './S3FilePicker'
 
+function resolveNameConflictRudely(name: string, attempt: number) {
+  const ext = extname(name)
+  const base = name.split(ext)[0]
+  return `${base} (conflict ${attempt})${ext}`
+}
+
+function resolveName(
+  name: string,
+  obj: Record<string, any>,
```

Comment:
this argument could probably have a more descriptive name

Reply:
It's here for keys only, so it doesn't matter what kind of object is here. Just some object with keys